libtelemetry/src/main/java/com/mapbox/services/android/telemetry/MapEventFactory.java

```diff
@@ -234,6 +235,12 @@ private void checkLoad(Event.Type type) {
     }
   }
 
+  private void checkUserId(String userId) {
+    if (userId == null && userId.isEmpty()) {
```

Comment:
This would cause a `NullPointerException` 💥 if `userId` is `null`.

> Method invocation 'isEmpty' may produce 'java.lang.NullPointerException'

The IDE hints you about this potential issue/crash.

Reply:
`IllegalArgumentException` should be thrown in two situations:
- If `userId` is `null`
- If `userId` is not `null` but it's empty

Why not using [`TelemetryUtils#isEmpty`](https://github.com/mapbox/mapbox-events-android/blob/master/libtelemetry/src/main/java/com/mapbox/services/android/telemetry/TelemetryUtils.java#L44)?